syntax = "proto2";

package braillel_rules;

message PartOfSpeech {
  required string name = 1;
  required bool before_space = 2;
  optional bool after_space = 3;
  optional bool allow_line_break = 4;
  optional string reading_word_match = 5;
  optional int reading_word_length = 6;
}

message Rule {
  required PartOfSpeech current_pos = 1;
  repeated PartOfSpeech next_pos = 2;
}

message BrailleRules {
  repeated Rule rule = 1;
}
